clients/cobol-lsp-vscode-extension/src/services/util/SubroutineUtils.ts

```diff
@@ -12,18 +12,34 @@
  *   Broadcom, Inc. - initial API and implementation
  */
 
-import { searchCopybookInWorkspace } from "./FSUtils";
+import { searchCopybookInExtensionFolder } from "./FSUtils";
 import { COBOL_EXT_ARRAY } from "../../constants";
 import { SettingsService } from "../Settings";
 
-/**
- * This function try to resolve a given subroutine by searching COBOL source file with the same name
- * in local file system in directories specified in settings.
- * @param name the name of subroutine
- * @return subroutine file URI if it was found or undefined otherwise
- */
-export function resolveSubroutineURI(name: string): string {
-  const folders: string[] | undefined =
-    SettingsService.getSubroutineLocalPath();
-  return searchCopybookInWorkspace(name, folders, COBOL_EXT_ARRAY)!;
+export class ResolveSubroutineUriService {
+  storagePath: string;
+  constructor(storagePath: string) {
+    this.storagePath = storagePath;
+  }
+
+  /**
+   * This function try to resolve a given subroutine by searching COBOL source file with the same name
+   * in local file system in directories specified in settings.
+   * @param name the name of subroutine
+   * @return subroutine file URI if it was found or undefined otherwise
+   */
+  public resolveSubroutineURI(name: string, storagePath: string): string {
+    const folders: string[] | undefined =
+      SettingsService.getSubroutineLocalPath();
+    return searchCopybookInExtensionFolder(
+      name,
+      folders,
+      COBOL_EXT_ARRAY,
+      storagePath,
+    )!;
+  }
+}
+
+export function resolveSubroutineUriHandler(this: any, name: string): string {
+  return this.resolveSubroutineURI(name, this.storagePath);
```

Comment:
The original function is exactly the same, it just needed an extra argument.

Reply:
how this extra argument could be provided by cobol/resolveSubroutine event ?

Reply:
```
function resolveSubroutineURI(storagePath: string, name: string){...}
resolveSubroutineURI.bind(undefined, "....");
```

Reply:
thanks. reverted here : f671879b9b58f97ef4705b2cf88b75d06bd23373